addon/index.js

```diff
@@ -17,43 +17,43 @@ export default Mixin.create({
     this._super(...arguments);
 
     setupRouter(this);
-  },
 
-  destroy() {
-    reset();
+    this.on('routeWillChange', () => {
```

Comment:
This change definitely is a breaking change (maybe thats fine?) as it only supports 3.6+

If you need to stay compatible, you would bring in ember-compatiblity-helpers and wrap these things in an `if`:

```js
import { gte } from 'ember-compatibility-helpers';

// ...snip...
if (gte('3.6.0-beta.1')) {
  // code for new APIs
} else {
  // code for older deprecated APIs
}
```

During a production build, the unused condition will be completely stripped (meaning there is no additional bloat to applications by having these conditionals).

Reply:
Thanks @rwjblue I will do that. It's odd that all our tests are passing on older Ember. They must not be truly testing things.